### PR TITLE
Backport of docs: Update docs for release of Consul API Gateway v0.3.0 into release/1.12.x

### PR DESCRIPTION
**Pull request by im2nguyen**
### Description
Describe why you're making this change, in plain English.

### Testing & Reproduction steps
* In the case of bugs, describe how to replicate
* If any manual tests were done, document the steps and the conditions to replicate
* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding

### Links
Include any links here that might be helpful for people reviewing your PR (Tickets, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.

Please be mindful not to leak any customer or confidential information. HashiCorp employees may want to use our internal URL shortener to obfuscate links.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

**Review by danielehc — APPROVED** [review]
Content LGTM. The preview does not build yet so could not check the look on UI, but markdown seems correct.

File: website/content/docs/api-gateway/tech-specs.mdx [hunk 1]
```diff
@@ -18,9 +18,10 @@ Verify that your environment meets the following requirements prior to using Con
 Your datacenter must meet the following requirements prior to configuring the Consul API Gateway:
 
 - Kubernetes 1.21+
+  - Kubernetes 1.24 is not supported at this time.
 - `kubectl` 1.21+
-- Consul 1.12.0+
-- HashiCorp Consul Helm chart 0.43.0+
+- Consul 1.11.2+
```
Comment (danielehc) [inline website/content/docs/api-gateway/tech-specs.mdx:23]: @im2nguyen is this the correct version? Is this not for Consul 1.12?

File: website/content/docs/release-notes/consul-api-gateway/v0_3_x.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  not be overridden. This hard limit will probably be increased in the future.
+
+- **Cross Namespace Reference Policies for Certificates:**
+  Reference policies is a security feature introduced in the 0.2 release. In
+  this release, Reference Policies are now required for TLS certificates when a
+  `listener` has a `certificateRef` that is in a different `namespace`.
+
+## Supported Software
+
+- Consul 1.11.2+
```
Comment (danielehc) [inline website/content/docs/release-notes/consul-api-gateway/v0_3_x.mdx:36]: @im2nguyen is this the correct version? Is this not for Consul 1.12?

Comment (im2nguyen) [reply website/content/docs/release-notes/consul-api-gateway/v0_3_x.mdx:36]: this is the correct version. this is taken from teh changes suggested by @Jeff-Apple 

https://github.com/hashicorp/consul/pull/13427/files

File: website/content/docs/api-gateway/tech-specs.mdx (hunk 1 above)
Comment (im2nguyen) [reply website/content/docs/api-gateway/tech-specs.mdx:23]: this is the correct version. this is taken from teh changes suggested by @Jeff-Apple 

https://github.com/hashicorp/consul/pull/13427/files

**Pull request merged**